rules/RULE_5_3_A_provide_doxygen_function_comment_on_function_in_header.py

```diff
@@ -131,13 +131,88 @@ def test5(self):
  void FunctionB() {
 }
 """)
-        assert  CheckErrorContent(__name__)
+        assert CheckErrorContent(__name__)
 
     def test6(self):
         self.Analyze("thisfile.h",
 """
 int a;
  void FunctionB();
 """)
-        assert  CheckErrorContent(__name__)
+        assert CheckErrorContent(__name__)
+
+    def test7(self):
+        self.Analyze("thisfile.h",
+"""
+/**
+ *
+ */
+extern void FunctionB();
+""")
+        assert not CheckErrorContent(__name__)
+
+    def test8(self):
+        self.Analyze("thisfile.h",
+"""
+class J {
+protected :
+    /** HELLO */
+    A();
+private :
+    B();
+    C() {
+    }
+}
+""")
+        assert not CheckErrorContent(__name__)
+
+    def test9(self):
+        self.Analyze("thisfile.h",
+"""
+///
+extern void FunctionB();
+""")
+        assert CheckErrorContent(__name__)
 
+    def test10(self):
+        self.Analyze("thisfile.h",
+"""
+class J {
+protected :
+    /// HELLO

```

Comment:
It would be nice if the file introduction contained the fact that ```///``` comments are also not valid. Tests 9-12 all are based on this.